syntax="proto3";
package codegen;
option go_package="github.com/zjaco13/sdks/go-sdk/codegen";

message AddResourceProviderRequest {
    string cluster_name = 1;
    string name = 2;
    ResourceProvider resource_provider = 3;
}

message ResourceProvider {
    oneof resource_provider {
        VpcProvider vpc_provider = 1;
    };

}
message AddVpcProviderRequest{
    string cluster_name = 1;
    string name = 2;
    VpcProvider vpc_provider = 3;
}

message VpcProvider{
    optional string vpcId = 1;

}
